src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/EnvelopeRepository.java

```diff
@@ -21,20 +21,21 @@
     /**
      * Finds envelope for a given container, zip file name and status.
      *
-     * @param container from where container originated
-     * @param zipFileName of envelope
-     * @param status of envelope
-     * @return Optional envelope
+     * @param container from where container originated.
+     * @param zipFileName of envelope.
+     * @param status of envelope.
+     * @return A list of envelopes.
      */
     @Query("select e from Envelope e"
         + " where e.container = :container"
         + "   and e.zipFileName = :zip"
         + "   and e.status = :status"
         + " order by e.createdAt desc"
     )
-    Optional<Envelope> checkLastEnvelopeStatus(
+    List<Envelope> findRecentEnvelopes(
```

Comment:
findRecentEnvelopesByStatus() or am I being pedantic again?

Reply:
@luigibk by container, zipfilename AND status if anything ;)

Reply:
Apparently having any kind of `By` message in there would confuse a lot since query have not only `status` to filter against. It was easier and still understandable to make just `recentEnvelopes` and you will have type-hint explicitly saying what you need: `container`, `zip`, `status`

Reply:
as longer as there's only single `find` were's good imho 👍